### ISSUE TITLE
setTimeout is not a function

### ISSUE THREAD

**Issue by simonepizzamiglio** [testing-library/native-testing-library#66]
- `react-native` or `expo`: `rect-native`
- `native-testing-library` version: `^4.0.14`
- `jest-preset`: `@testing-library/react-native`
- `react-native` version: `0.60.5`
- `node` version: `10.14.2`

### Relevant code or config:

TextSwitcher.js
```js
import React, { useEffect, useState } from 'react';
import { Text, View, Animated } from 'react-native';

const animationDuration = 300;

function TextSwitcher({ isOn, onText, offText }) {
	const [onTextOpacity] = useState(new Animated.Value(0));
	const [offTextOpacity] = useState(new Animated.Value(1));

	useEffect(
		() => {
			Animated.parallel([
				Animated.timing(onTextOpacity, {
					toValue: isOn ? 1 : 0,
					duration: animationDuration,
					delay: isOn ? animationDuration : 0,
				}).start(),
				Animated.timing(offTextOpacity, {
					toValue: isOn ? 0 : 1,
					duration: animationDuration,
					delay: isOn ? 0 : animationDuration,
				}).start(),
			]);
		},
		[isOn]
	);

	return (
		<View>
			<Animated.Text style={{ opacity: onTextOpacity}}>
				{onText}
			</Animated.Text>
			<Animated.Text style={{ opacity: offTextOpacity}}>
				{offText}
			</Animated.Text>
		</View>
	);
}


export default TextSwitcher;

```

TextSwitcher.test.js
```js
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, wait } from '@testing-library/react-native';
import TextSwitcher from './TextSwitcher';

it('if on, shows onText', async () => {
	const { getByText } = render(
		<TextSwitcher isOn={true} offText={offText} onText={onText} />
	);
	await wait(() => {
		expect(getByText(onText)).toBeVisible();
	});
});
```

### What you did:

I created a component that switches 2 texts based on a boolean value

### What happened:

When I run the test, it returns `setTimeout is not a function`

### Problem description:

Since the fade-in and fade-out animations of the text when it switches have a delay, I need to test it within an async function.

I followed the example in the example in the documentation
https://www.native-testing-library.com/docs/guide-disappearance#waiting-for-disappearance

When I run the test, it returns this error `setTimeout is not a function`

**Comment by bcarroll22** [comment]
Hmm I'm not sure what would be causing this and I'm not sure how it could be an issue with this library, unfortunately. The `wait` helper uses the default global `setTimeout` function which is included in Node. Your version looks good, my best guess would be to check if you've got any sort of global set up in the project.

Could you post the stack trace? Maybe that would help pinpoint where the issue is occurring.

**Comment by ioss** [comment]
While I am also not sure how you are able to get `setTimeout is not a function`.  As @bcarroll22 wrote a stack trace might help, or even better a simple demo to reproduce this.
I suspect you somehow "managed" to have a global `window` variable, without `setTimeout`, in which case a demo would be better to find out how it got there.

Apart from that `@testing-library/jest-dom` is not meant to be used with react-native, as it needs DOM-elements and `native-testing-library` does not provide these. (Maybe you added something, to simulate a browser environment, with a `window`, but without `window.setTimeout`?)

Anyway to test if an element exists, remove `@testing-library/jest-dom` (and anything else providing dom/browser emulation) and use for example:
```javascript
await wait(() => {
  expect(getByText(onText)).toBeTruthy();
});
```
as documented in the link provided by you.

**Comment by simonepizzamiglio** [comment]
Here is the stack trace
```
at node_modules/wait-for-expect/lib/index.js:57:5
at tryCallTwo (node_modules/promise/lib/core.js:45:5)
at doResolve (node_modules/promise/lib/core.js:200:13)
at new Promise (node_modules/promise/lib/core.js:66:3)
at waitForExpect (node_modules/wait-for-expect/lib/index.js:34:10)
at wait (node_modules/@testing-library/react-native/dist/lib/wait.js:21:10)
at node_modules/@testing-library/react-native/dist/lib/wait.js:30:17
at _callee$ (node_modules/@testing-library/react-native/dist/preset/configure.js:26:57)
at tryCatch (node_modules/regenerator-runtime/runtime.js:45:40)
at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:271:22)
```

**Comment by bcarroll22** [comment]
hi @simonepizzamiglio did you attempt the steps provided by @ioss?

**Comment by simonepizzamiglio** [comment]
Hi @bcarroll22 yes I did, this is my package.json jest configuration,I thought it might be related to that `window: {}`, but if I removed it I get `ReferenceError: window is not defined`.
```
"jest": {
  "preset": "@testing-library/react-native",
  "setupFilesAfterEnv": [
    "jest-extended",
    "./__mocks__/firebase.js",
    "@testing-library/react-native/cleanup-after-each"
  ],
  "transformIgnorePatterns": [
    "/node_modules/(?!native-base)/"
  ],
  "globals": {
    "window": {}
  }
}
```

**Comment by bcarroll22** [comment]
Got it. The `window: {}` _is_ likely causing an issue because removing it changes the error. So it looks like you'll have to do some better mocking in order to make this work with your exact use case.

In jest-native and this library, we don't ever reference the window, so I don't think it's specifically an issue within one of them. Just to be sure, I updated `wait-for-expect` to the latest version which is included in the release `^4.1.0`.

I don't know that I'll be able to be much further help unfortunately since I don't have experience with `youtube-iframe` ☹️ but what I can say is that it looks like that module is designed for the web, not React Native. React Native doesn't have a window object unless you're creating a fake one there too, which will probably cause heartache down the line at some point.

Good luck and I hope you get it figured out soon! I'm going to go ahead and close this because I'm confident saying this isn't an issue with our library, but rather a mixture of other things. Sorry I can't be more help.

**Comment by simonepizzamiglio** [comment]
I fixed it, there is no `youtube-iframe` library in my `package.json`, but there is in the `yarn.lock`.
I deleted my `yarn.lock file`, run` yarn install`, removed `window: {}` and everything works fine.

Someone probably installed the library `youtube-iframe` and didn't remove it properly, thanks for you help :)
